impeller/toolkit/glvk/trampoline.cc

```diff
@@ -0,0 +1,328 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/toolkit/glvk/trampoline.h"
+
+#include <array>
+
+#include "flutter/fml/closure.h"
+#include "flutter/fml/trace_event.h"
+#include "impeller/base/timing.h"
+#include "impeller/base/validation.h"
+#include "impeller/geometry/point.h"
+#include "impeller/renderer/backend/gles/formats_gles.h"
+#include "impeller/toolkit/android/proc_table.h"
+#include "impeller/toolkit/egl/image.h"
+
+namespace impeller::glvk {
+
+static GLuint kAttributeIndexPosition = 0u;
+static GLuint kAttributeIndexTexCoord = 1u;
+
+static constexpr const char* kVertShader = R"IMPELLER_SHADER(#version 100
+
+precision mediump float;
+
+attribute vec2 aPosition;
+attribute vec2 aTexCoord;
+
+varying vec2 vTexCoord;
+
+void main() {
+  gl_Position = vec4(aPosition, 0.0, 1.0);
+  vTexCoord = aTexCoord;
+
+}
+)IMPELLER_SHADER";
+
+static constexpr const char* kFragShader = R"IMPELLER_SHADER(#version 100
+
+#extension GL_OES_EGL_image_external : require
+
+precision mediump float;
+
+uniform samplerExternalOES uTexture;
+uniform mat4 uSTTransformation;
```

Comment:
I am pretty sure I have used `(u, v)` and `(s, t)` interchangeably for a long long time. But, I think there is a very subtle difference between the two. The spec. mentions `(s, t)` when using coordinates that sample from a texture. But these need not be in the 0.0 to 1.0 range. Depending on the wrap mode, the conversion between `(s, t)` and `(u, v)` may end up differently. For instance, 1.25 as `s` may end up being 0.25 `u` in repeat-wrap or 0.75 `u` in mirror-wrap. However, if you use `(s, t)` for say deriving the min or mag filter in the implementation, things might go very wrong. Hence, I think, the distinction. While for us, needing to be this specific is usually being pedantic. 

The docs for [getTransformMatrix](https://developer.android.com/reference/android/graphics/SurfaceTexture#getTransformMatrix(float[])) mentioned `(s, t)` which I am using directly and am just depending on the wrap modes to do the rest. So I just use the same terminology. But, I am never giving the operation coordinates outside the unit range pre-transformation and the docs say the contents post transformation outside the range are undefined. So I am pretty sure the coordinates post transformation won't be outside the unit range. So perhaps UV is also a fine terminology. Would you like me to use that instead to avoid confusion?

Speaking of wrap modes, I realized I should at least default it to something instead of depending on whatever Android specified on the bind. Will fix that bit.

Reply:
I renamed it to UV since that seems be our convension.